packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4036,13 +4036,11 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     assert(slots == null || newWidgets.length == slots.length);
 
     Element? replaceWithNullIfForgotten(Element child) {
-      return forgottenChildren != null && forgottenChildren.contains(child) ? null : child;
+      return (forgottenChildren?.contains(child) ?? false) ? null : child;
     }
 
     Object? slotFor(int newChildIndex, Element? previousChild) {
-      return slots != null
-          ? slots[newChildIndex]
-          : IndexedSlot<Element?>(newChildIndex, previousChild);
+      return slots?[newChildIndex] ?? IndexedSlot<Element?>(newChildIndex, previousChild);
```

Comment:
I wonder if this potentially changes semantics because the new version resolves to creating an `IndexedSlot` even if `slots` is not null but `slots[newChildIndex]` is null.

Reply:
That's a really good point! I should probably go ahead and revert this one, thanks :)